components/Banner/src/MDCBannerView.m

```diff
@@ -558,11 +558,40 @@ - (void)updateBannerFont {
 }
 
 - (void)updateTextFont {
-  UIFont *textFont = self.textLabel.font;
-  if (self.mdc_adjustsFontForContentSizeCategory && textFont.mdc_scalingCurve) {
-    textFont = [textFont mdc_scaledFontForTraitEnvironment:self];
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    NSAttributedString *attributedText = self.textLabel.attributedText;
+    NSMutableAttributedString *mutableAttributedText = [attributedText mutableCopy];
+    UIFont *textFont = self.textLabel.font;
+    if (textFont.mdc_scalingCurve) {
+      textFont = [textFont mdc_scaledFontForTraitEnvironment:self];
+    }
+    self.textLabel.font = textFont;
+    [mutableAttributedText beginEditing];
+    __block BOOL hasScalableFont = NO;
+    __weak typeof(self) weakSelf = self;
+    [mutableAttributedText
+        enumerateAttribute:NSFontAttributeName
+                   inRange:NSMakeRange(0, mutableAttributedText.length)
+                   options:0
+                usingBlock:^(id value, NSRange range, BOOL *stop) {
+                  if (value) {
+                    UIFont *previousFont = (UIFont *)value;
+                    if (previousFont.mdc_scalingCurve) {
+                      hasScalableFont = YES;
+                      UIFont *scaledFont =
+                          [previousFont mdc_scaledFontForTraitEnvironment:weakSelf];
```

Comment:
Please capture `weakSelf` in a strong pointer so it isn't lost while enumerating.

Reply:
This is the only place `self` is referenced in this block. Capturing this weak pointer is not needed in this case.

Reply:
Discussed offline (`mdc_scaledFontForTraitEnvironment` only takes nonnull). Addressed in 5ed3632d0ba4658c7ba55f653cbd2b38dee6d895.